src/org/traccar/smpp/SmppClient.java

```diff
@@ -184,31 +193,41 @@ private void destroySession() {
         }
     }
 
-    public synchronized void sendMessageSync(String destAddress, String message) throws RecoverablePduException,
-            UnrecoverablePduException, SmppTimeoutException, SmppChannelException, InterruptedException {
+    public synchronized void sendMessageSync(String destAddress, String message, boolean command)
+            throws RecoverablePduException, UnrecoverablePduException, SmppTimeoutException, SmppChannelException,
+            InterruptedException, IllegalStateException {
         if (getSession() != null && getSession().isBound()) {
-            byte[] textBytes = CharsetUtil.encode(message, charsetName);
-
             SubmitSm submit = new SubmitSm();
+            byte[] textBytes;
+            if (command) {
+                textBytes = CharsetUtil.encode(message, commandsCharsetName);
+                submit.setDataCoding(commandsDataCoding);
+            } else {
+                textBytes = CharsetUtil.encode(message, notificationsCharsetName);
+                submit.setDataCoding(notificationsDataCoding);
+            }
+            submit.setShortMessage(textBytes);
             submit.setSourceAddress(new Address(sourceTon, sourceNpi, sourceAddress));
             submit.setDestAddress(new Address(destTon, destNpi, destAddress));
-            submit.setDataCoding(dataCoding);
-            submit.setShortMessage(textBytes);
             SubmitSmResp submitResponce = getSession().submit(submit, submitTimeout);
-            Log.debug("SMS submited, msg_id: " + submitResponce.getMessageId());
+            if (submitResponce.getCommandStatus() == SmppConstants.STATUS_OK) {
+                Log.debug("SMS submited, msg_id: " + submitResponce.getMessageId());
+            } else {
+                throw new IllegalStateException(submitResponce.getResultMessage());
```

Comment:
In some cases communication with smpp server can be successful, but really something wrong. For example destination address. We need to log this.